connectivity/cellular/source/framework/AT/AT_CellularContext.cpp

```diff
@@ -757,9 +758,10 @@ void AT_CellularContext::check_and_deactivate_context()
     // For EPS, if an attempt is made to disconnect the last PDN connection, then the MT responds with ERROR
     if (_is_context_active && (rat < CellularNetwork::RAT_E_UTRAN || active_contexts_count > 1)) {
         _at.clear_error();
-        deactivate_context();
     }
 
+    deactivate_context();
```

Comment:
The comment above _For EPS, if an attempt is made to disconnect the last PDN connection, then the MT responds with ERROR..._  comes from 3GPP TS 27.007.

Unlike EPS, NB-IoT can be without PDN so could be fixed above like `if (_is_context_active && (rat < CellularNetwork::RAT_E_UTRAN || rat < CellularNetwork::RAT_NB1 || active_contexts_count > 1)) {`.

Reply:
Added this. Does this also apply to CAT-M1?

Reply:
> Added this. Does this also apply to CAT-M1?

Good question. According to changes it doesn't. But I'm pretty sure it should because CAT M1 and NB1 have similar working concepts from fw point of view, both needing contexts.

So maybe instead of:
` if (_is_context_active && (rat < CellularNetwork::RAT_E_UTRAN || rat == CellularNetwork::RAT_NB1 || active_contexts_count > 1)) {`
use:
` if (_is_context_active && (rat < CellularNetwork::RAT_UNKNOWN || active_contexts_count > 1)) {`